### PR TITLE
My Jetpack: Update data handling - Implement request status in Product Card

### PR DESCRIPTION
**Pull request by retrofox**
Fixes #

#### Changes proposed in this Pull Request:

This PR updates the data handling implementation for the Products;

* Change the response shape for the `/site/products/$product` endpoint
* Update Redux implementation according to it
* Implement the initial requesting status for the Product card when an action is run

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Go to 'My Jetpack dashboard
* Play with the Backup card. You should see the error when trying to deactivate the product. It's expected since the handler is not implemented yet on the server-side.
* You can see how the data is propagated using network and redux tool of your dev tools, in the client:

https://user-images.githubusercontent.com/77539/150874627-2e734bfd-d910-4db8-bce2-aaad9f36a4c0.mov

You can hardcode the response for both for the `activate_product()` and `detivate_product()` methods respectively

```php
$deactivate_product_result = $product;
$deactivate_product_result['status'] = 'active';
return rest_ensure_response( $deactivate_product_result, 200 );
```

```php
$deactivate_product_result = $product;
$deactivate_product_result['status'] = 'inactive';
return rest_ensure_response( $deactivate_product_result, 200 );
```

for the `detivate_product()` method

https://user-images.githubusercontent.com/77539/150891154-64500c82-7b2f-4fb4-92e5-67a7ed985c1f.mov

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/state/reducers.js [hunk 1]
```diff
@@ -9,13 +9,29 @@ import { combineReducers } from '@wordpress/data';
...
+			return {
+				...state,
+				isFetching: {
+					...state.isFetching,
+					[ productId ]: isFetching,
+				},
+				errors: {
+					...state.errors,
+					[ productId ]: isFetching ? undefined : state.errors[ productId ],
+				},
```
Comment (renatoagds) [inline projects/packages/my-jetpack/_inc/state/reducers.js:31]: This is starting to become really complex, mainly if we continue to create one key for each property/flag, would say that prefer to wrap everything inside product slug as key:

```javascript
products = {
  boost: {
    isFeching: false,
    errors: {},
    ...productData
  }
}
```

**Review by renatoagds — CHANGES_REQUESTED** [review]

File: projects/packages/my-jetpack/_inc/state/reducers.js (hunk 1 above)
Comment (retrofox) [reply projects/packages/my-jetpack/_inc/state/reducers.js:31]: I wouldn't like to pollute the item with these data that don't belong to the product by itself. The good thing about that is we can just pick the item data and pass them to the component, making the data propagation simpler and transparent, imo.

About complexity, I don't see significant pros and cons in between both approaches.

```
products: {
	backup: {
    	name,
		description,
		// more product data,
		isFetching: true,
		error: {
			code:...,
			message: ...,
			status: {}
		},
	},
	boost: {
		...
	},
	crm: {
		....
	}
}
```

```
products: {
	items: {
		backup: {
			name,
			description,
			// more product data,
		},
		boost: {
			...
		},
		crm: {
			...
		}
	}
	isFetching: {
		backup: true,
	},
	errors: {
		backup: {
			code:...,
			message: ...,
			status: {}
		},
	}
}
```

Comment (renatoagds) [reply projects/packages/my-jetpack/_inc/state/reducers.js:31]: You're right, looks the same :)

Comment (renatoagds) [reply projects/packages/my-jetpack/_inc/state/reducers.js:31]: Yes! Rephrasing, they look similar.

**Review by renatoagds — APPROVED** [review]
LGTM. Just remembered that I'm not the best one to review the PHP side, so could be good for another person looking at it.

**Comment by retrofox** [comment]
> LGTM. Just remembered that I'm not the best one to review the PHP side, so could be good for another person looking at it.

Thanks for your review. I think we are ok since @leogermani took and added some changes here, too.

**Pull request merged**